Added additional ability for Entity and Finder

Djaconda opened the pull request:
- to have access to changed attributes within afterSave event for Entity
- to call DataSource functions that work with attributes for Entity (getOldAttributes, getDirtyAttributes etc...)
- to call Query functions that returns finally result for Finder (column, scalar etc..)

dmitrybukavin committed:
Added additional ability for Entity and Finder

- to have access to changed attributes within afterSave event for Entity
- to call DataSource functions that work with attributes for Entity (getOldAttributes, getDirtyAttributes etc...)
- to call Query functions that returns finally result for Finder (column, scalar etc..)

prowwid commented:
The key point of splitting ActiveReccord to Repository, DataMapper, Record and Entity is to split DB, Support and Domain logick across those elements and let them be responsible only for the single purpose.

Adding such methods kicks us back to ActiveRecord so I agains the current implementation. 

In my opinion all the functionality that added to Entity must go to Repository. E.g. if you need to refresh Entity, you just use `$repository->refresh($entity)` and so on with other methods.
```
$repository->isJustAdded($entity);
$repository->isJustUpdated($entity);
and so on
```

It is possible that for releseale of V1 of the library we might drop direct relation between DataMapper and entity having Repository fully managing storage action and Entity having only domain related functionality.

dmitrybukavin committed:
Added additional ability for EntitiesRepository to get access for changed entity attributes

prowwid commented on src/DB/EntitiesRepository.php:
```diff
@@ -96,6 +96,81 @@ public function validate(contracts\DomainEntity $entity) {
+    //endregion
+
+    //region ----------------------- ENTITY DATA METHODS --------------------------
+    public function isNewOrJustAdded(contracts\DomainEntity $entity): bool {
```
For some reason namespaces in this file are in lower case (including "domain" in the top of the file). Please edit it to the upper case (Contracts)

dmitrybukavin committed:
yii2-domain-27: fixed namespace of contracts

----
The add action doesn't redirect to edit page after save entity when redirect url is not set

prowwid committed:
Merge branch 'master' into changed-attributes
